federatedscope/core/auxiliaries/utils.py

```diff
@@ -446,3 +447,39 @@ def format_dict(target_dict):
     elif isinstance(hooks_set, dict):
         print_obj = format_dict(hooks_set)
     return json.dumps(print_obj, indent=2).replace('\n', '\n\t')
+
+
+def get_resource_info(filename):
+    if filename is None or not os.path.exists(filename):
+        logger.info('The device information file is not provided')
+        return None
+
+    # Users can develop this loading function according to resource_info_file
+    # As an example, we use the device_info provided by FedScale (FedScale:
+    # Benchmarking Model and System Performance of Federated Learning
+    # at Scale), which can be downloaded from
+    # https://github.com/SymbioticLab/FedScale/blob/master/benchmark/dataset/
+    # data/device_info/client_device_capacity The expected format is
+    # { INDEX:{'computation': FLOAT_VALUE_1, 'communication': FLOAT_VALUE_2}}
+    with open(filename, 'br') as f:
+        device_info = pickle.load(f)
+    return device_info
+
+
+def calculate_time_cost(instance_number,
+                        model_size,
```

Comment:
`model_size` can be a more general one such as `comm_size`

Reply:
Done